modules/ROOT/pages/apikit-workflow-convert-existing.adoc

```diff
@@ -1,104 +1,239 @@
+	</db:config>
+    <flow name="apiFlow">
+		<http:listener config-ref="HTTP_Listener_config" path="/api/*"/>
+		<apikit:router config-ref="Router"/>
+	</flow>
+	<flow name="GET-Users-FLOW">
+		<db:select config-ref="Database_Config">
+			<db:sql ><![CDATA[SELECT * FROM USERS]]></db:sql>
+		</db:select>
+	</flow>
+	<flow name="GET-User-By-Id-FLOW">
+		<db:select config-ref="Database_Config" >
+			<db:sql ><![CDATA[SELECT * FROM USERS WHERE :userIdParameter == userID]]></db:sql>
+			<db:input-parameters ><![CDATA[#[{userIdParameter: attributes.queryParams.id}]]]></db:input-parameters>
+		</db:select>
+	</flow>
+</mule>
+----
+
+== Scaffolding Approach
+
+Follow the process at xref:apikit-workflow-convert-existing.adoc#step-1-design-your-api-specification[]. When completed, include an alternative workflow for scaffolding your API specification.
+
+=== Step 2: Scaffold Your API Specification
+
+Scaffold your API to auto-generate the main, console, and resources flows:
+
+. In *Package Explorer*, right-click `/src/main/resources/api/api.raml`.
+. Select *Mule*.
+. Click *Generate Flows from Local REST API*.
+
+A new XML file with the same name as the API is created containing the scaffolded flows.
+
+NOTE: With this approach you get the APIkit router configuration, the error handler, and the console flow out of the box. You also get a flow that maps each resource.
+
+=== Step 3: Reference Flows
+
+At this point, you have some options:
+
+* Delete the scaffolded flows for each resource and map the original flows as mentioned in xref:apikit-workflow-convert-existing.adoc#step-3:-map-api-resources-actions-to-flows
+[].
+* Remove any operations from the scaffolded flows and add a flow reference instead. By choosing this option you introduce an additional layer of indirection, which can be inconvenient in some scenarios.
+
+[source,xml,linenums]
+----
+<mule>
+...
+    <flow name="get:\users\userbyid:api-config">
+		<flow-ref name="GET-User-By-Id-FLOW"/>
+    </flow>
+    <flow name="get:\users:api-config">
+		<flow-ref name="GET-Users-FLOW"/>
+    </flow>
+</mule>
+----
+
+* Transfer the content of the existing flows to the scaffolded flows that adhere to the routing naming convention. If the application is extensive, performing these modifications can be a laborious task.
+
+[source,xml,linenums]
+----
+<mule>
+...
+    <flow name="get:\users:api-config">
+		<db:select config-ref="Database_Config">
+			<db:sql ><![CDATA[SELECT * FROM USERS]]></db:sql>
+		</db:select>
+	</flow>
+	<flow name="get:\users\userbyid:api-config">
+		<db:select config-ref="Database_Config" >
+			<db:sql ><![CDATA[SELECT * FROM USERS WHERE :userIdParameter == userID]]></db:sql>
+			<db:input-parameters ><![CDATA[#[{userIdParameter: attributes.queryParams.id}]]]></db:input-parameters>
+		</db:select>
+	</flow>
+</mule>
+----
+
+Consider the size and complexity of the application when deciding on the best approach.
+
+== Test Your API
+
```

Comment:
Can you give an example of a call that does this?

Reply:
Requesting assistance to SMEs.
Most probably, it's OK to send the reader to this [API testing](https://dev-docs-internal.kqa.msap.io/beta-apikit-new-content-structure/apikit/4.x/apikit-4-implement-rest-api#test-your-api-implementation).
Awaiting SME's confirmation.

Reply:
Yes, it is. Done!
Thanks!